arm64: dts: qcom: msm8939-asus-z00t: Add initial device tree

erikas9987 opened the pull request:
Asus ZenFone 2 Laser/Selfie (Z00T) is a series of smartphones released in 2015, featuring a 1080p panel and a Qualcomm Snapdragon 615 (MSM8939) chipset. This PR brings up initial mainline support for the following variants of Asus ZenFone 2 Laser/Selfie:
- ZE551KL
- ZD551KL
- ZE601KL (note: this device appears to feature an MSM8939v2 instead)
- ZX550KL (mentioned in the downstream code, yet the existence of such device could not be confirmed)

Functionality was tested on Asus ZenFone Selfie (ZD551KL). The results are provided in the following table.

| Component | Working? | Model | Notes |
| --- | --- | --- | --- |
| USB | Yes | N/A | USB is configured using an ID pin. |
| Internal storage | Yes | N/A | |
| SD card | Yes | N/A | lk2nd fails to boot mainline Linux from an SD card. |
| Buttons | Yes | N/A | Volume up, down and power key. |
| WiFi | Partial | WCN3660B | The signal is much weaker than if the chipset was used in Android. |
| Graphics | Partial | Adreno 405 | Hardware video acceleration support is missing, as Venus for MSM8939 is WIP. |
| Panel | Yes | tm5p5-nt35596 | Display flickers when switching brightness over the user interface. Switching brightness over sysfs works. In this PR, the panel is configured with `qcom,dsi-phy-regulator-ldo-mode` enabled, however, in a similar device that also has `qcom,regulator-ldo-mode`  enabled downstream, Z00L, such property is missing. This PR also introduces a new driver for this panel, however, a driver for it is already present in mainline [here](https://github.com/msm8916-mainline/linux/blob/msm8916/6.17-rc6/drivers/gpu/drm/panel/panel-asus-z00t-tm5p5-n35596.c) |
| Touchscreen | Yes | Focaltech FT5306 | |
| Accelerometer | Yes | Invensense MPU8515 | |
| Magnetometer | Broken | Asahi Kasei AK09911 | Device does not show up in `iio-sensor-proxy` and reports very minor value changes after being moved around. |
| Ambient light sensor | Broken | Capella CM36283/CM36686 | ~~This device is missing drivers in the mainline kernel and schematics for it are not publicly available.~~ WIP driver for this sensor can be found [here.](https://github.com/erikas9987/linux/tree/cm36283) |
| Hall Sensor | Untested | N/A | This PR does not introduce support for the hall sensor as it could not be confirmed that it is working. However, device tree bindings for the hall sensor are available in a separate branch [here](https://github.com/erikas9987/linux/tree/hall-sensor) |
| Audio | ~~Yes~~ Partial | NXP TFA9887 | Built-in microphone, speakers, headphone jack output are working as expected. External microphone was not tested.  ZE601KL uses a custom audio amplifier named NXP TFA9887, for which there is no support in mainline, however, a datasheet is publicly available [here](https://www.nxp.com/docs/en/data-sheet/TFA9887_SDS.pdf) |
| Modem | Yes | N/A | At the time of bringing up the device, modem was broken on MSM8939. However, modem was confirmed to be operational after applying patches listed [here.](https://github.com/msm8916-mainline/linux/pull/413) |
| Camera | Broken | N/A | qcom-camss subsystem is not supported by MSM8939. Mainline drivers for the flash LED are also missing, however, a datasheet for it, although a lacking one, is publicly available [here](https://www.mouser.com/datasheet/2/472/SKY81296_PS_203109B-1102094.pdf). |
| NFC | N/A | N/A | This device does not support NFC. |
| Battery | ~~Partial~~ Broken | Li-Poly | This device relies on pm8916 vm-bms to receive battery readings downstream. On mainline, they are inaccurate. For now, battery nodes are removed from this PR, however, they remain available [here](https://github.com/erikas9987/linux/tree/battery) |
| Charger | Broken | Summit smb358 | A driver for the charger IC is present in mainline, however, the configuration of it in this device's downstream device tree is different than what's seen in other devices using this charger. In particular, it is configured using `qcom,otg-gpio` and `qcom,inok-gpio` GPIOs, while a mainline driver expects an interrupt or a usb-vbus regulator. |

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@9743013
Signed-off-by: lmdpdg <lmdpdg@localhost>

erikas9987 committed:
arm64: dts: qcom: msm8939-asus-z00t: add panel

This commit adds support for the panel used in Asus ZenFone 2
Laser/Selfie. The panel used in this device is tm5p5-nt35596.

Signed-off-by: Erikas Bitovtas <xerikasxx@gmail.com>

a-andre commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+		bias-disable;
+	};
+
+	lcd_bl_en_default: lcd-bl-en-default-state {
```
This and `mdss_{default,sleep}` should be part of the "Add panel" commit. See for example https://github.com/msm8916-mainline/linux/commit/5c2ceaf9b9097c3e29b2eafe88db0180b10f55a9

And
```
&gpu {
	status = "okay";
};
```
should probably be moved to the "Add panel" commit, too.

erikas9987 replied:
Done.

a-andre commented:
Not necessary for this pull request, but when sending upstream, you should add a commit that adds `asus,z00t` to [`qcom.yaml`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/arm/qcom.yaml?h=v6.17-rc7#n185). Note, this bindings commit should be first in your series.

erikas9987 commented:
> Not necessary for this pull request, but when sending upstream, you should add a commit that adds `asus,z00t` to [`qcom.yaml`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/arm/qcom.yaml?h=v6.17-rc7#n185). Note, this bindings commit should be first in your series.

A separate commit for the binding in documentation, or squashed into the one for the device tree?

a-andre commented:
A separate commit, see for example https://lore.kernel.org/linux-arm-msm/20230727-bq_m5-v3-0-df5e227c79d0@apitzsch.eu/

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_keys_default>;
```
```suggestion
		pinctrl-0 = <&gpio_keys_default>;
		pinctrl-names = "default";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+
+		startup-delay-us = <200>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&sd_vmmc_en_default>;
```
```suggestion
		pinctrl-0 = <&sd_vmmc_en_default>;
		pinctrl-names = "default";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+		vid-supply = <&pm8916_l6>;
+
+		reset-gpios = <&tlmm 112 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&mag_reset_default>;
```
```suggestion
		pinctrl-0 = <&mag_reset_default>;
		pinctrl-names = "default";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+		vdd-supply = <&pm8916_l8>;
+		vddio-supply = <&pm8916_l6>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&imu_default>;
```
```suggestion
		pinctrl-0 = <&imu_default>;
		pinctrl-names = "default";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+		touchscreen-size-x = <1080>;
+		touchscreen-size-y = <1920>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&touchscreen_default>;
```
```suggestion
		pinctrl-0 = <&touchscreen_default>;
		pinctrl-names = "default";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+};
+
+&mdss_dsi0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
```
```suggestion
	pinctrl-0 = <&mdss_default>;
	pinctrl-1 = <&mdss_sleep>;
	pinctrl-names = "default", "sleep";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+		backlight-gpios = <&tlmm 21 GPIO_ACTIVE_HIGH>;
+		power-supply = <&pm8916_l8>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&lcd_bl_en_default>;
```
```suggestion
		pinctrl-0 = <&lcd_bl_en_default>;
		pinctrl-names = "default";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+};
+
+&blsp_uart2 {
+	status = "okay";
+	pinctrl-0 = <&blsp_uart2_console_default>;
+	pinctrl-1 = <&blsp_uart2_console_sleep>;
+	pinctrl-names = "default", "sleep";
```
```suggestion
	pinctrl-0 = <&blsp_uart2_console_default>;
	pinctrl-1 = <&blsp_uart2_console_sleep>;
	pinctrl-names = "default", "sleep";
	status = "okay";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+
+&pm8916_bms {
+	status = "okay";
+	monitored-battery = <&battery>;
```
```suggestion
	monitored-battery = <&battery>;
	status = "okay";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+};
+
+&sdhc_2 {
+	status = "okay";
+	vmmc-supply = <&reg_sd_vmmc>;
+
+	pinctrl-0 = <&sdc2_default &sdc2_cd_default>;
+	pinctrl-1 = <&sdc2_sleep &sdc2_cd_default>;
+	pinctrl-names = "default", "sleep";
+	cd-gpios = <&tlmm 38 GPIO_ACTIVE_LOW>;
```
```suggestion
	vmmc-supply = <&reg_sd_vmmc>;

	pinctrl-0 = <&sdc2_default &sdc2_cd_default>;
	pinctrl-1 = <&sdc2_sleep &sdc2_cd_default>;
	pinctrl-names = "default", "sleep";
	cd-gpios = <&tlmm 38 GPIO_ACTIVE_LOW>;
	status = "okay";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,351 @@
+};
+
+&sound {
+	status = "okay";
+	audio-routing =
+		"AMIC1", "MIC BIAS External1",
+		"AMIC2", "MIC BIAS Internal2",
+		"AMIC3", "MIC BIAS External1";
```
```suggestion
	audio-routing =
		"AMIC1", "MIC BIAS External1",
		"AMIC2", "MIC BIAS Internal2",
		"AMIC3", "MIC BIAS External1";
	status = "okay";
```

erikas9987 replied:
Shall I squash the changes into the initial device tree commit?

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,394 @@
+		compatible = "invensense,mpu6515";
+		reg = <0x68>;
+
+		interrupt-parent = <&tlmm>;
+		interrupts = <36 IRQ_TYPE_EDGE_RISING>;
```
```suggestion
		interrupts-extended = <&tlmm 36 IRQ_TYPE_EDGE_RISING>;
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,394 @@
+		compatible = "edt,edt-ft5306";
+		reg = <0x38>;
+
+		interrupt-parent = <&tlmm>;
+		interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
```
```suggestion
		interrupts-extended = <&tlmm 13 IRQ_TYPE_EDGE_FALLING>;
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,394 @@
+		bias-pull-up;
+	};
+};
+
```
```suggestion
```

a-andre commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,324 @@
+		pinctrl-0 = <&gpio_keys_default>;
+		pinctrl-names = "default";
+
+		label = "GPIO Buttons";
```
This label could be removed. IMO a hall sensor, which might be added later to `gpio-keys`, is not a "GPIO button".

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,346 @@
+&mdss_dsi0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
```
```suggestion
	pinctrl-0 = <&mdss_default>;
	pinctrl-1 = <&mdss_sleep>;
	pinctrl-names = "default", "sleep";
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8939-asus-z00t.dts:
```diff
@@ -0,0 +1,346 @@
+		power-supply = <&pm8916_l8>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&lcd_bl_en_default>;
```
```suggestion
		pinctrl-0 = <&lcd_bl_en_default>;
		pinctrl-names = "default";
```

erikas9987 committed:
UPSTREAM: dt-bindings: arm: qcom: Add Asus ZenFone 2 Laser/Selfie

Add a compatible for Asus ZenFone 2 Laser/Selfie (1080p)

Signed-off-by: Erikas Bitovtas <xerikasxx@gmail.com>
Acked-by: Rob Herring (Arm) <robh@kernel.org>
Link: https://lore.kernel.org/r/20251008182106.217340-2-xerikasxx@gmail.com
Signed-off-by: Bjorn Andersson <andersson@kernel.org>

erikas9987 committed:
UPSTREAM: arm64: dts: qcom: msm8939-asus-z00t: add initial device tree

Add an initial device tree for Asus ZenFone 2 Laser/Selfie. This
includes support for:
- UART
- USB
- Internal storage
- MicroSD
- Volume keys
- Touchscreen: Focaltech FT5306
- Accelerometer: Invensense MPU6515
- Magnetometer: Asahi Kasei AK09911
- Vibrator
- Audio input and output
- Modem

Signed-off-by: Erikas Bitovtas <xerikasxx@gmail.com>
Reviewed-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Link: https://lore.kernel.org/r/20251008182106.217340-3-xerikasxx@gmail.com
Signed-off-by: Bjorn Andersson <andersson@kernel.org>

erikas9987 commented:
@TravMurav https://lore.kernel.org/phone-devel/176160465208.73268.15556003473353414195.b4-ty@kernel.org/

TravMurav commented:
Thanks!
